packages/flutter_tools/test/integration.shard/test_utils.dart

```diff
@@ -165,18 +162,27 @@ class ProcessResultMatcher extends Matcher {
   @override
   bool matches(dynamic item, Map<dynamic, dynamic> matchState) {
     final ProcessResult result = item as ProcessResult;
-
-    if (stdoutSubstring != null) {
-      _foundStdout = (result.stdout as String).contains(stdoutSubstring!);
-      matchState['stdout'] = result.stdout;
+    bool foundStdout = true;
+    bool foundStderr = true;
+
+    final String stdout = result.stdout as String;
+    final String stderr = result.stderr as String;
+    if (stdoutPattern != null) {
+      foundStdout = stdout.contains(stdoutPattern!);
```

Comment:
Very cool, i was going to loop back and make it work for regex patterns but didn't know it was this easy!